### PR TITLE
Explore by example

### PR DESCRIPTION
**Pull request by isabelge**
**Explore-by-Example**

This branch supports Explore-by-Example an interactive dataset exploration.
(There is a second pull request for Polypheny-UI part.)

**Frontend:**
- Explore-by-Example Page
- entry selection with true and false
- shows decision tree
- switch between Weka Classification and creation of SQL query
- show created SQL query

**Backend:**
- added Weka for classification 
- ExploreManager for handling Explore Elements
- classification with J48 decision Tree of labeled data
- translation from Weka to SQL

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -86,6 +86,8 @@ dependencies {
 
     api group: "org.apache.calcite", name: "calcite-linq4j", version: calcite_linq4j_version
     api group: "org.apache.calcite.avatica", name: "avatica-core", version: avatica_core_version
+
+    compile group: 'nz.ac.waikato.cms.weka', name: 'weka-stable', version: '3.8.0'
```
Comment (vogti) [inline core/build.gradle]: Please add a variable (`weka_version`) for the weka version in the `gradle.properties`.

File: webui/src/main/java/org/polypheny/db/webui/Crud.java [hunk 2]
```diff
@@ -703,6 +709,96 @@ Result insertRow( final Request req, final Response res ) {
...
+        try {
+            Result r = executeSqlSelect( transaction, queryExplorationRequest, query ).setInfo( new Debug().setGeneratedQuery( query ) );
+
+            for ( int i = 0; i < r.getHeader().length; i++ ){
+                typeInfo.put( r.getHeader()[i].name, r.getHeader()[i].dataType );
+               group.add( r.getHeader()[i].dataType );
+            }
+
+        } catch (  QueryExecutionException | RuntimeException e ) {
+            e.printStackTrace();
```
Comment (vogti) [inline webui/src/main/java/org/polypheny/db/webui/Crud.java]: Please remove printStackTrace() calls.

File: statistic/src/main/java/org/polypheny/db/statistic/exploreByExample/Explore.java [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+        unlabeled.setClassIndex( unlabeled.numAttributes() - 1 );
+        Instances labeled = new Instances( unlabeled );
+
+        for ( int i = 0; i < unlabeled.numInstances(); i++ ) {
+            double clsLabel = 0;
+
+            try {
+                clsLabel = tree.classifyInstance( unlabeled.instance( i ) );
+            } catch ( Exception e ) {
+                log.error("classification of unlabeled data not possible");
```
Comment (vogti) [inline statistic/src/main/java/org/polypheny/db/statistic/exploreByExample/Explore.java]:
```suggestion
                log.error("classification of unlabeled data not possible", e);
```

File: webui/src/main/java/org/polypheny/db/webui/Crud.java [hunk 4]
```diff
@@ -703,6 +709,96 @@ Result insertRow( final Request req, final Response res ) {
...
+            }
+        } catch ( QueryExecutionException | TransactionException | RuntimeException e ) {
+            log.error( "Caught exception while executing a query from the console", e );
+            executionTime += System.nanoTime() - temp;
+            result = new Result( e ).setInfo( new Debug().setGeneratedQuery( query ) );
+            results.add( result );
+            try {
+                transaction.rollback();
+            } catch ( TransactionException ex ) {
+                log.error( "Caught exception while rollback", e );
```
Comment (vogti) [inline webui/src/main/java/org/polypheny/db/webui/Crud.java]:
```suggestion
                log.error( "Caught exception while rollback", ex );
```

File: core/build.gradle (hunk 1 above)
Comment (isabelge) [reply core/build.gradle]: resolved with commit [89aff06](url)

Comment (isabelge) [reply core/build.gradle]: resolved in commit: [https://github.com/polypheny/Polypheny-DB/pull/211/commits/89aff06943a24226e43ad9d8007bd1415d395264](url)

File: webui/src/main/java/org/polypheny/db/webui/Crud.java (hunk 2 above)
Comment (isabelge) [reply webui/src/main/java/org/polypheny/db/webui/Crud.java]: resolved in commit:[https://github.com/polypheny/Polypheny-DB/pull/211/commits/5aa89f0c4a4106cfc45536101cf1877310f5dea9](url)

File: statistic/src/main/java/org/polypheny/db/statistic/exploreByExample/Explore.java (hunk 3 above)
Comment (isabelge) [reply statistic/src/main/java/org/polypheny/db/statistic/exploreByExample/Explore.java]: resolved in commit:[https://github.com/polypheny/Polypheny-DB/pull/211/commits/5aa89f0c4a4106cfc45536101cf1877310f5dea9](url)

File: webui/src/main/java/org/polypheny/db/webui/Crud.java (hunk 4 above)
Comment (isabelge) [reply webui/src/main/java/org/polypheny/db/webui/Crud.java]: resolved with commit: [https://github.com/polypheny/Polypheny-DB/pull/211/commits/a4b6fc9d10a26bb12387f463dbd9a964e2bfae50](url)

**Pull request merged**